### PR TITLE
🐛 Remove coupling of search/collides input geometry type to tree item geometry type

### PR DESCRIPTION
**Pull request by twelch**
TLDR:
This PR resolves a bug where if you pass a more specific Geometry type to the rbush constructor of the Features you will insert into the tree, like `new rbush<Point>()`, it is incorrectly enforcing that the geometry passed as input to the `search` and `collides` methods also be a Point, but it should be able to be a Polygon or any other valid Geometry.

Unblocks this PR - https://github.com/Turfjs/turf/pull/2225#discussion_r762602015

Background:
The geojson-rbush package, among other things, adda a generic type parameter to the rbush constructor, which by default enforces that the geometry type of items inserted into the rbush tree is a GeoJSON `Geometry`. 
```
declare class RBush<G extends Geometry, P extends GeoJsonProperties> {
```

Usually the default is fine.  But a generic parameter can be overridden, allowing you to specify a narrower Geometry type if you know for example that you will only insert type `Point` for example.  By doing this, when you use rbush methods to fetch items from your tree, you automatically get back tree items with the narrower geometry type you expect through the magic of Typescript generics, so that you don't have to cast.

To confirm this decoupling is valid, I can see that all that `geojson-rbush.search` does, it gets the bbox of your search geometry, and then calls rbush.search with that bbox.  Same for collides.  I think that the type of this input geometry to these two methods can/should be completely independent of the geometry type of items in the tree
```
    tree.search = function (geojson) {
        var features = rbush.prototype.search.call(this, this.toBBox(geojson));
        return featureCollection(features);
    };
```

### CODE REVIEW

**Review by JamesLMilner — APPROVED** [review]

**Comment by twelch** [comment]
@JamesLMilner a new version can be published when you are ready
